### ISSUE TITLE
"No resource files found on the classpath" when using parallelStream()

### ISSUE THREAD

**Issue by nimo23** [OpenGamma/Strata#2144]
I get the following error:

```
09:27:21,319 SEVERE [com.opengamma.strata.collect.io.ResourceConfig] 
No resource files found on the classpath: 
[META-INF/com/opengamma/strata/config/base/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base1/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base2/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base3/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base4/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base5/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base6/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base7/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base8/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base9/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library1/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library2/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library3/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library4/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library5/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library6/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library7/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library8/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library9/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application1/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application2/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application3/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application4/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application5/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application6/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application7/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application8/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application9/HolidayCalendar.ini]
09:27:21,322 SEVERE [com.opengamma.strata.collect.named.ExtendedEnum] 
Failed to load ExtendedEnum for interface com.opengamma.strata.basics.date.HolidayCalendar: java.lang.IllegalStateException:
No files found on the classpath: 
[META-INF/com/opengamma/strata/config/base/HolidayCalendar.ini,
META-INF/com/opengamma/strata/config/base1/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base2/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base3/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base4/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base5/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base6/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base7/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base8/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/base9/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library1/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library2/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library3/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library4/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library5/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library6/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library7/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library8/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/library9/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application1/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application2/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application3/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application4/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application5/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application6/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application7/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application8/HolidayCalendar.ini, 
META-INF/com/opengamma/strata/config/application9/HolidayCalendar.ini]
	at deployment.test.war//com.opengamma.strata.collect.io.ResourceConfig.orderedResources0(ResourceConfig.java:274)
	at deployment.test.war//com.opengamma.strata.collect.io.ResourceConfig.lambda$orderedResources$0(ResourceConfig.java:239)
	at deployment.test.war//com.opengamma.strata.collect.Unchecked.wrap(Unchecked.java:107)
	at deployment.test.war//com.opengamma.strata.collect.io.ResourceConfig.orderedResources(ResourceConfig.java:239)
	at deployment.test.war//com.opengamma.strata.collect.io.ResourceConfig.combinedIniFile(ResourceConfig.java:175)
	at deployment.test.war//com.opengamma.strata.collect.named.ExtendedEnum.of(ExtendedEnum.java:143)
	at deployment.test.war//com.opengamma.strata.basics.date.HolidayCalendars.<clinit>(HolidayCalendars.java:72)
	at deployment.test.war//com.opengamma.strata.basics.StandardReferenceData.<clinit>(StandardReferenceData.java:26)
	at deployment.test.war//com.opengamma.strata.basics.ReferenceData.standard(ReferenceData.java:59)
```

The error occurs if I use java lamda style code in combination with `parallelStream()`. When using java `stream()` or a normal java for-loop instead of `parallelStream()` all works as expected and no error occurs. 

_I use Strata v2.7.0, JDK 12._

**Comment by jodastephen** [comment]
Have you got some sample code that fails. I can't find anything immediately wrong. thanks

**Comment by nimo23** [comment]
For example, the following code works and no error is thrown:

```
// this fails because of parallelStream()
Set<Boolean> check{
// using stream()-version
return ids.stream().flatMap(id -> check(id).stream()).collect(Collectors.toSet());
}

boolean check(HolidayCalendarId id){
  return id.resolve(ReferenceData.standard()).isHoliday(LocalDate.now());
}
```

The following code does NOT work and throws the exception:

```
// this fails because of parallelStream()
Set<Boolean> check{
// using parallelStream()-version
return ids.parallelStream().flatMap(id -> check(id).stream()).collect(Collectors.toSet());
}

boolean check(HolidayCalendarId id){
  return id.resolve(ReferenceData.standard()).isHoliday(LocalDate.now());
}
```

The stacktrace when using `parallelStream()`-version:

```
11:23:49,731 SEVERE [com.opengamma.strata.collect.io.ResourceConfig] 
No resource files found on the classpath: 
[META-INF/com/opengamma/strata/config/base/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base1/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base2/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base3/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base4/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base5/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base6/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base7/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base8/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base9/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library1/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library2/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library3/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library4/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library5/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library6/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library7/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library8/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library9/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application1/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application2/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application3/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application4/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application5/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application6/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application7/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application8/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application9/HolidayCalendar.ini]

11:23:49,734 SEVERE [com.opengamma.strata.collect.named.ExtendedEnum] Failed to load ExtendedEnum for interface com.opengamma.strata.basics.date.HolidayCalendar: java.lang.IllegalStateException: 
No files found on the classpath: [META-INF/com/opengamma/strata/config/base/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base1/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base2/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base3/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base4/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base5/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base6/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base7/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base8/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/base9/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library1/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library2/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library3/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library4/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library5/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library6/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library7/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library8/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/library9/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application1/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application2/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application3/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application4/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application5/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application6/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application7/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application8/HolidayCalendar.ini, META-INF/com/opengamma/strata/config/application9/HolidayCalendar.ini]
..
1:23:49,737 ERROR com.opengamma.strata.basics.ReferenceDataNotFoundException: 
Reference data not found for identifier 'NYSE' of type 'HolidayCalendarId'
	at test.war//com.opengamma.strata.basics.ReferenceData.getValue(ReferenceData.java:111)
	at test.war//com.opengamma.strata.basics.date.HolidayCalendarId.resolve(HolidayCalendarId.java:227)
```

**Comment by jodastephen** [comment]
Running this Java code on Java 8 or 11 with 38,000 IDs results in no error:

```
  public static void main(String[] args) {
    check2();
    System.out.println("done");
  }

  static Set<Boolean> check2() {
    return IDS.parallelStream().flatMap(id -> Stream.of(check(id))).collect(Collectors.toSet());
  }

  static boolean check(HolidayCalendarId id) {
    return id.resolve(ReferenceData.standard()).isHoliday(LocalDate.now());
  }
```

Are you using Java? (Your original code did not compile).

I imagine there is a race condition somewhere, but it is hard to examine without a reproducer

**Comment by nimo23** [comment]
> Are you using Java?
Yes, I am using Java 12. I tried to extract the critical code into two simple methods to show what I called in essence. Was not really a reproducer. Your code works for me as well. Actually, it is hard to trace the reason for this error because all I get is the stacktrace which shows errors coming from Strata. 


>I imagine there is a race condition somewhere

I will try to find the reason and let you know..I need some time..
